docs/content/guide/filter.ngdoc

```diff
@@ -109,7 +109,7 @@ as the first argument. Any filter arguments are passed in as additional argument
 function.
 
 The filter function should be a [pure function](http://en.wikipedia.org/wiki/Pure_function), which
-means that it should be stateless and idempotent, and not rely for example on other Angular services.
+means that it should be deterministic and not rely, for example, on other Angular services.
```

Comment:
`deterministic` actually does cover the idea of `stateless` since all stateless functions are inherently deterministic. However, it's also quite possible to have stateful functions that are also deterministic as long as the state is entirely self-contained. This is the reason I changed it.